src/libstd/panic.rs

```diff
@@ -144,10 +147,28 @@ pub trait RefRecoverSafe {}
 /// // });
 ///
 /// // This, however, will compile due to the `AssertRecoverSafe` wrapper
+/// let result = panic::recover(AssertRecoverSafe::new(|| {
+///     variable += 3;
+/// }));
+/// // ...
+/// ```
+///
+/// Sometimes this behavior may be a bit too coarse grained, however, and
```

Comment:
I think this comment is kind of vague. I would maybe say something like:

Wrapping the entire closure amounts to a blanket assertion that all variables are recover safe. This has the downside that if new captures are added in the future, they will also be considered recover safe. Therefore, you may prefer to just wrap individual capture, as shown below. This is more annotation, but it ensures that if a new upvar is added which is not recover safe, you will get a compilation error at that time, which will allow you to consider whether that new capture in fact represent a bug or not.